Agent: consistent 'containerID' field in the log of the requests EP-delete and EP-create

romanspb80 opened the pull request:
Previously, it was difficult to find creates and deletes based on container ID due to non-consistent container ID field in the log: 'EP create' used short-form container ID, 'EP delete' used standart form. For examle:

'EP create':
level=info msg="New endpoint" containerID=77ce9db8c9 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=3100 identity=5 ipv4=172.18.12.151 ipv6= k8sPodName=default/pod subsys=endpoint

vs

'EP delete':
level=info msg="Delete endpoint request" id="container-id:7eab59b2e768e9845af3988be879558186e3215e018577c139a3405cbd424e37" subsys=daemon

This patch fixes it: 'EP delete' uses short-form container ID as 'EP create'.

Fixes: #13658

Signed-off-by: Roman Ptitcyn <romanspb@yahoo.com>

sayboras commented on daemon/cmd/endpoint.go:
```diff
@@ -709,6 +709,14 @@ func (d *Daemon) DeleteEndpoint(id string) (int, error) {
 	} else {
+		containerID := ep.GetShortContainerID()
+		if containerID != "" {
+			log.WithField("containerID", containerID).Info("Delete endpoint request")
```
maybe containerID ->  logfields.ContainerID for consistency.

romanspb80 replied:
OK. Thanks!

sayboras commented on daemon/cmd/endpoint.go:
```diff
@@ -709,6 +709,14 @@ func (d *Daemon) DeleteEndpoint(id string) (int, error) {
+			log.WithField("containerID", containerID).Info("Delete endpoint request")
+		} else {
+			// if the containerID does not exist, then the IP address is inserted in the log
+			log.WithField("IPv4", ep.GetIPv4Address()).WithField("IPv6", ep.GetIPv6Address()).Info(
```
same as above, IPv4 -> logfields.IPv4, IPv6 -> logfields.IPv6.

romanspb80 replied:
This is a good idea. I'll do it.

sayboras commented on daemon/cmd/endpoint.go:
```diff
@@ -709,6 +709,14 @@ func (d *Daemon) DeleteEndpoint(id string) (int, error) {
+		} else {
+			// if the containerID does not exist, then the IP address is inserted in the log
+			log.WithField("IPv4", ep.GetIPv4Address()).WithField("IPv6", ep.GetIPv6Address()).Info(
+			"Delete endpoint request")
```
gofmt failure here

romanspb80 replied:
No. It works

romanspb80 replied:
> gofmt failure here

Sorry. I misunderstood this issue.

sayboras reviewed (commented):
Any reason why the below line is still there ? I am under assumption that this should be removed/updated as mentioned in github issue.

https://github.com/cilium/cilium/blob/4c827bf7d1e2c0f2a5ff4c6b606f7c17bda316fa/daemon/cmd/endpoint.go#L702-L703

christarazi commented on daemon/cmd/endpoint.go:
```diff
@@ -709,6 +710,19 @@ func (d *Daemon) DeleteEndpoint(id string) (int, error) {
 		return 0, api.Error(DeleteEndpointIDInvalidCode, err)
 	} else {
+		containerID := ep.GetShortContainerID()
+		msg := "Deleting endpoint"
```
Let's keep the log msg the same as it was: `Delete endpoint request`

christarazi commented on daemon/cmd/endpoint.go:
```diff
@@ -709,6 +710,19 @@ func (d *Daemon) DeleteEndpoint(id string) (int, error) {
+			log.WithFields(logrus.Fields{
+				logfields.ContainerID: containerID,
+			}).Info(msg)
+		}
```
```suggestion
		msg := "Deleting endpoint"
		switch containerID := ep.GetShortContainerID(); containerID {
		case "":
		        log.WithFields(logrus.Fields{
		    		logfields.IPv4:   ep.GetIPv4Address(),
				logfields.IPv6:   ep.GetIPv6Address(),
			}).Info(msg)
		default:
			log.WithFields(logrus.Fields{
				logfields.ContainerID: containerID,
			}).Info(msg)
		}
```

romanspb80 replied:
OK
Done.
Thanks

christarazi reviewed (changes requested):
Thanks for the PR! Change looks good overall, a few minor nits. As Tam pointed out, seems like the linter is failing on `gofmt`. Also, please squash your commits into one.

christarazi commented on daemon/cmd/endpoint.go:
```diff
@@ -699,8 +699,9 @@ func (d *Daemon) deleteEndpointQuiet(ep *endpoint.Endpoint, conf endpoint.Delete
-	log.WithField("id", id).Info("Delete endpoint request")
+	// The log message about request "DELETE /endpoint/{id} request" with level=debug has been
+	// in the previous step (deleteEndpointID Handle).
+	// So now needs a log message (level=info) with the information about EP to be deleted.
```
I'm not sure I understand the need for these comments?

romanspb80 replied:
I think too. I made them as an update.
So I've removed them.

christarazi reviewed (commented):
One more minor comment. Also, there are a few typos in the commit msg: `examle` and `standart`. Otherwise, this is good to go for me.

romanspb80 committed:
Agent: consistent 'containerID' field in the log of the requests EP-delete and EP-create

Previously, it was difficult to find creates and deletes based on container ID due to non-consistent container ID field in the log: 'EP create' used short-form container ID, 'EP delete' used standart form. For examle:

'EP create':
level=info msg="New endpoint" containerID=77ce9db8c9 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=3100 identity=5 ipv4=172.18.12.151 ipv6= k8sPodName=default/pod subsys=endpoint

vs

'EP delete':
level=info msg="Delete endpoint request" id="container-id:7eab59b2e768e9845af3988be879558186e3215e018577c139a3405cbd424e37" subsys=daemon

This patch fixes it: 'EP delete' uses short-form container ID as 'EP create'.

Fixes: #13658

Signed-off-by: Roman Ptitcyn <romanspb@yahoo.com>

sayboras reviewed (approved):
LGTM :100:
